src/main/java/io/cdap/plugin/aws/s3/source/S3BatchSource.java

```diff
@@ -202,6 +209,44 @@ public void validate(FailureCollector collector) {
       }
     }
 
+    private void validatePath(FailureCollector collector) {
+      if (!containsMacro("path") && (!path.startsWith("s3a://") && !path.startsWith("s3n://"))) {
+        collector.addFailure("Path must start with s3a:// or s3n://.", null).withConfigProperty(NAME_PATH);
+      }
+      if (!containsMacro("path") && path.startsWith("s3n://") && !Strings.isNullOrEmpty(connection.getSessionToken())) {
+        collector.addFailure("Temporary credentials are only supported for s3a:// paths.", null)
+          .withConfigProperty(NAME_PATH);
+      }
+    }
+
+    private void validateCredentials(FailureCollector collector) {
+
+      AmazonS3 s3 = getS3Client(Regions.DEFAULT_REGION.getName());
+      try {
+        s3.listBuckets();
+      } catch (AmazonS3Exception exception) {
+        String errorMessage = exception.getErrorMessage();
+        collector.addFailure(String.format("Invalid values: %s.", errorMessage),
+                             "Please provide valid values.").withStacktrace(exception.getStackTrace());
+      }
+    }
+
+    private AmazonS3 getS3Client(String region) {
+      AmazonS3ClientBuilder builder = AmazonS3ClientBuilder.standard();
+      if (!Strings.isNullOrEmpty(region)) {
+        builder.setRegion(region);
+      }
+      AWSCredentials creds;
+      if (connection.getSessionToken() == null) {
+        creds = new BasicAWSCredentials(connection.getAccessID(), connection.getAccessKey());
```

Comment:
This is not correct I guess even though connection is not macro which we checked above but connector configs can still be macro so we cannot generate credentials until they are not macro.

Reply:
ref: https://github.com/data-integrations/amazon-s3-plugins/blob/b49756d7ed89ee9f297f01c85f8d1f91b94e2a7c/src/main/java/io/cdap/plugin/aws/s3/common/S3ConnectorConfig.java#L117

Reply:
Moved the validation code inside the `ConnectorConfig` class. This should take care of the issue you highlighted. dbaf2f1183f1c59b081da9170b80d6ef24acc0f7